lib/absinthe/ecto.ex

```diff
@@ -94,11 +104,15 @@ defmodule Absinthe.Ecto do
   field :author, :user, resolve: assoc(MyApp.Repo, :author)
   ```
   """
-  def assoc(repo, association) do
-    fn parent, _, _ ->
+  def assoc(repo, association, query_fun) do
+    fn parent, args, %{context: ctx} ->
       case Map.get(parent, association) do
         %Ecto.Association.NotLoaded{} ->
-          ecto_batch(repo, parent, association)
+          if query_fun != nil do
+            ecto_batch(repo, parent, {association, fn query -> query_fun.(query, args, ctx) end})
```

Comment:
Have you had a chance to see if this actually works? The last time I tried to use anonymous functions in the batch key it failed because anonymous functions are never equal to one another, so no batching ever actually happened:

```elixir
iex(1)> x = fn -> 1 end
#Function<20.52032458/0 in :erl_eval.expr/5>
iex(2)> y = fn -> 1 end
#Function<20.52032458/0 in :erl_eval.expr/5>
iex(3)> x == y
false
```